cli/cli/commands/run/run.go

```diff
@@ -608,6 +624,23 @@ func parseVerbosityFlag(flags *flags.ParsedFlags) (command_args_run.Verbosity, e
 	return verbosity, nil
 }
 
+// Get the image download flag is present, and parse it to a valid ImageDownload value
+func parseImageDownloadFlag(flags *flags.ParsedFlags) (kurtosis_core_rpc_api_bindings.ImageDownloadMode, error) {
+
+	defaultImageMode := kurtosis_core_rpc_api_bindings.ImageDownloadMode_missing
+
+	imageDownloadStr, err := flags.GetString(imageDownloadFlagKey)
+	if err != nil {
+		return defaultImageMode, stacktrace.Propagate(err, "An error occurred getting the image-download using flag key '%s'", imageDownloadFlagKey)
+	}
+	imageDownload, err := command_args_run.ImageDownloadString(imageDownloadStr)
+	if err != nil {
+		return defaultImageMode, stacktrace.Propagate(err, "Invalid image-download value: '%s'. Possible values are %s", imageDownloadStr, strings.Join(command_args_run.ImageDownloadStrings(), ", "))
```

Comment:
I see that if this returns, `err` you return the `defaultImageMode` with an `err` but then above you check for the error, then propagate the error, so the `defaultImageMode` actually never ends up being used afaict.

Reply:
If I'm right, I suggest getting rid of the `defaultImageMode`, and then return `nil, stacktrace.Prop...` in case of an error, and then have the default image download value be configured as the default in `StarlarkRunConfig`, and be overrided by `starlark_run_config.WithImageDownloadMode(imageDownload)`. This way we only have to reason about the default image download mode in one place `StarlarkRunConfig` instead of two.

Reply:
No good apart from `nil` breaking the code because is the type is not a pointer. I can make it a pointer but not sure the tradeoffs.